### PR TITLE
mlflow.evaluate(): allow arbitrary positive labels

### PR DESCRIPTION
**Pull request by dbczumar**
Signed-off-by: dbczumar <corey.zumar@databricks.com>

## What changes are proposed in this pull request?

mlflow.evaluate(): allow arbitrary positive labels

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/models/evaluation/default_evaluator.py [hunk 1]
```diff
@@ -1065,12 +1065,9 @@ def _generate_model_predictions(self):
...
 
             if self.is_binomial:
-                if list(self.label_list) not in [[0, 1], [-1, 1]]:
-                    raise ValueError(
-                        "Binary classifier evaluation dataset positive class label must be 1 or"
-                        " True, negative class label must be 0 or -1 or False, and dataset"
-                        " must contains both positive and negative examples."
-                    )
+                if self.pos_label in self.label_list:
+                    self.label_list.remove(self.pos_label)
```
Comment (BenWilson2) [inline mlflow/models/evaluation/default_evaluator.py:1069]:
```suggestion
                    self.label_list = np.delete(self.label_list, np.where(self.label_list == self.pos_label))
```
I think numpy returns a copy of the immutable array and deletion is performed through index position location.

Comment (dbczumar) [reply mlflow/models/evaluation/default_evaluator.py:1069]: Great catch! ndarray doesn't even define `remove()`. Not sure how this was succeeding. Will revert the other 2.0 PR and fix this :)

Comment (BenWilson2) [reply mlflow/models/evaluation/default_evaluator.py:1069]: I'm working on a branch for 2.0

Comment (BenWilson2) [reply mlflow/models/evaluation/default_evaluator.py:1069]: almost ready

File: mlflow/models/evaluation/default_evaluator.py [hunk 2]
```diff
@@ -1065,12 +1065,9 @@ def _generate_model_predictions(self):
             self.is_binomial = self.num_classes <= 2
 
             if self.is_binomial:
-                if list(self.label_list) not in [[0, 1], [-1, 1]]:
```
Comment (bbarnes52) [inline mlflow/models/evaluation/default_evaluator.py:1068]: should this logic still execute if pos_label is not specified?

Comment (dbczumar) [reply mlflow/models/evaluation/default_evaluator.py:1068]: I think it's too restrictive in any case.

**Comment by dbczumar** [comment]
@mlflow-automation autoformat

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Pull request merged**